File: 算法/104. 二叉树的最大深度/MaxDepth.java
public class MaxDepth {


    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode() {
        }

        TreeNode(int val) {
            this.val = val;
        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    class Solution {

        public int maxDepth(TreeNode root) {
            if (root == null) {
                return 0;
            }
            int leftLen = 0;
            int rightLen = 0;
            if (root.left != null) {
                leftLen = maxDepth(root.left);
            }
            if (root.right != null) {
                rightLen = maxDepth(root.right);
            }
            return Math.max(leftLen, rightLen) + 1;
        }


    }

}

File: 算法/104. 二叉树的最大深度/MaxDepth.go
package _04__二叉树的最大深度

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}
	leftMax := maxDepth(root.Left)
	rightMax := maxDepth(root.Right)
	return max(leftMax, rightMax) + 1
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
